server/src/main/java/org/elasticsearch/index/mapper/AbstractScriptFieldType.java

```diff
@@ -156,12 +159,16 @@ private String unsupported(String query, String supported) {
         );
     }
 
-    protected final void checkAllowExpensiveQueries(SearchExecutionContext context) {
+    protected final void checkQueryScriptContext(SearchExecutionContext context) {
         if (context.allowExpensiveQueries() == false) {
             throw new ElasticsearchException(
                 "queries cannot be executed against runtime fields while [" + ALLOW_EXPENSIVE_QUERIES.getKey() + "] is set to [false]."
             );
         }
+        if (isResultDeterministic == false) {
+            // disable request cache
+            context.getClient();
```

Comment:
Do you think it'd be worth adding a method named `disableRequestCache` so you don't have to leave this comment?